src/main/java/com/commercetools/sync/inventories/InventorySync.java

```diff
@@ -168,119 +177,186 @@ public InventorySyncStatistics getStatistics() {
+                    throw new SyncProblemException();
+                })
+                .thenApply(existingSupplyChannels -> existingSupplyChannels.stream()
+                            .collect(toMap(Channel::getKey, Channel::getId)))
+                .thenCompose(supplyChannelKeyToId -> createMissingSupplyChannels(drafts, supplyChannelKeyToId));
     }
 
     /**
      * When {@code ensureChannel} from {@link InventorySyncOptions} is set to {@code true} then attempts to create
      * missing supply channels. Missing supply channel is a supply channel of key that can not be found in CTP project,
-     * but occurs in {@code drafts} list. Method returns {@link CompletionStage} of {@link Void} that indicates all
-     * possible creation attempts progress.
+     * but occurs in {@code drafts} list. Method returns {@link CompletionStage} of {@link Map} that contains updated
+     * {@code supplyChannelKeyToId}.
      *
      * @param drafts {@link List} containing {@link InventoryEntryDraft} objects where missing supply channels can occur
-     * @return {@link CompletionStage} of {@link Void} that indicates all possible creation attempts progress
+     * @param supplyChannelKeyToId mapping of supply channel key to supply channel Id for supply channels existing in
+     *                             CTP project.
+     * @return {@link CompletionStage} of {@link Map} that contains updated {@code supplyChannelKeyToId}
      */
     @SuppressFBWarnings("NP_NONNULL_PARAM_VIOLATION") // https://github.com/findbugsproject/findbugs/issues/79
-    private CompletionStage<Void> createMissingSupplyChannels(@Nonnull final List<InventoryEntryDraft> drafts) {
+    private CompletionStage<Map<String, String>> createMissingSupplyChannels(@Nonnull final List<InventoryEntryDraft>
+                                                                                     drafts,
+                                                                             @Nonnull final Map<String, String>
+                                                                                 supplyChannelKeyToId) {
         if (syncOptions.shouldEnsureChannels()) {
             final List<String> missingChannelsKeys = drafts.stream()
-                .map(SkuChannelKeyTuple::of)
-                .map(SkuChannelKeyTuple::getKey)
+                .map(ChannelKeyExtractor::extractChannelKey)
                 .distinct()
                 .filter(Objects::nonNull)
                 .filter(key -> !supplyChannelKeyToId.containsKey(key))
                 .collect(toList());
-            final List<CompletableFuture<Void>> creationStages = missingChannelsKeys.stream()
+            final List<CompletableFuture<Channel>> creationStages = missingChannelsKeys.stream()
                 .map(this::createMissingSupplyChannel)
+                .map(newChannelStage -> newChannelStage
+                    .handle((newChannel, exception) -> {
```

Comment:
Yes, if some channel failed to be created then some inventories might fail to sync after that.